src/controllers/room.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from 'express';
+import Room from '../models/Room';
+
+export async function createRoom(req: Request, res: Response) {
+    const { name, schedule, closed } = req.body;
+    if (typeof name !== 'string' || typeof closed !== 'boolean' || !Array.isArray(schedule)) {
+        res.status(400).json({ err: 'validation failed' });
```

Comment:
Let's change the JSON key to `error` for consistency with auth endpoints (and hopefully everything after) - this applies for all of your errors